### PR TITLE
Splitted content.media_capture setting into dedicated settings for audio, video and both

### PR DESCRIPTION
**Pull request by ykahveci**
Splitted `content.media_capture` into `content.media_audio_capture`, `content.media_audio_video_capture` and `content.media_video_capture`.

This is not only my first contribution to a project as big as this one, but it is also my first time touching python, so please double check everything.

By the way, this codebase is incredibly clean and easy to understand even by a PHP programmer.

### CODE REVIEW

**Comment by ykahveci** [comment]
I'll try. This failed check infuriates me.

**Comment by ykahveci** [comment]
My line seems to be too long

File: qutebrowser/browser/webengine/webenginetab.py [hunk 1]
```diff
@@ -900,9 +900,9 @@ class _WebEnginePermissions(QObject):
     _options = {
         0: 'content.notifications',
         QWebEnginePage.Geolocation: 'content.geolocation',
-        QWebEnginePage.MediaAudioCapture: 'content.media_capture',
-        QWebEnginePage.MediaVideoCapture: 'content.media_capture',
-        QWebEnginePage.MediaAudioVideoCapture: 'content.media_capture',
+        QWebEnginePage.MediaAudioCapture: 'content.media.audio_capture',
+        QWebEnginePage.MediaVideoCapture: 'content.media.video_capture',
+        QWebEnginePage.MediaAudioVideoCapture: 'content.media.audio_video_capture',
```
Comment (The-Compiler) [inline qutebrowser/browser/webengine/webenginetab.py:905]:
```suggestion
        QWebEnginePage.MediaAudioVideoCapture:
            'content.media.audio_video_capture',
```

to break the long line

File: tests/unit/config/test_configfiles.py [hunk 2]
```diff
@@ -617,6 +617,18 @@ def test_fonts_tabs(self, yaml, autoconfig):
...
+        val = 'ask'
+        autoconfig.write({'content.media_capture': {'global': val}})
+
+        yaml.load()
+        yaml._save()
+
+        data = autoconfig.read()
+        assert data['content.media.audio_capture']['global'] == val
+        assert data['content.media.audio_video_capture']['global'] == val
+        assert data['content.media.video_capture']['global'] == val
```
Comment (The-Compiler) [inline tests/unit/config/test_configfiles.py:630]: Could probably make this a bit nicer with a for-loop:

```python
        for setting in ['content.media.audio_capture',
                        'content.media.audio_video_capture',
                        'content.media.video_capture']:
            assert data[setting]['global'] == val
```

Comment (ykahveci) [reply tests/unit/config/test_configfiles.py:630]: Again I don't really speak python, but that looks good to me. I'll commit it

Comment (ykahveci) [reply tests/unit/config/test_configfiles.py]: Done!

**Comment by ykahveci** [comment]
> Looks good to me now! Going to bed, but plan to merge this tomorrow.

Great! Good night! 🌙

**Comment by mschilli87** [comment]
Honestly, I would prefer to keep `content.media_capture` but with options `none`, `audio`, `video`, or `both`: It reduces the  number of config options, stresses the common factor of capturing (note that `content.media.audio_capture` and `content.media.video_capture` only share the `content.media` prefix and all `content.media` options are about media *capture* without this being indicated by the category name), and the cnofig system would already support this as-is.
The only draw-back I see is that code enabling/disabling say audio capture has to know to switch to `both` when set to `video` and `audio` otherwise etc.

**Comment by ykahveci** [comment]
But we need an ask option as not every user wants to de facto allow or deny the permission. One could implement this with a migration from ask->ask allow->both deny->none I think. But again, I am a python beginner

**Comment by The-Compiler** [comment]
I think it's better to keep those as separate settings. I can see how a single setting seems appealing (that's also why I originally implemented this as a single setting, after all :laughing:), but in the end that's more trouble than it's worth.

Even if we had the values `none`, `audio`, `video`, `both` and `ask`, then answering "always" to e.g. an audio prompt would presumably change this from `ask` to `audio` for the given page. But then if the same page later asked for video permissions, it's unclear what the correct behavior would be. If requesting video permissions with the setting set to `audio` would always ask, then there would be no way to say "I want this website to get audio permissions, but never ask me about video anymore".

All in all, I think the benefits (and clarity) of having three different settings is bigger than the drawbacks, so I'm going to merge this as-is. Thanks @ykahveci for the contribution and @mschilli87 for the ideas (even if we didn't end up going that route)!

**Comment by ykahveci** [comment]
okay (yay)

**Pull request merged**